pcli/src/command/view/tx.rs

```diff
@@ -176,19 +179,35 @@ impl TxCmd {
     pub fn offline(&self) -> bool {
         false
     }
-    pub async fn exec<V: ViewClient>(&self, fvk: &FullViewingKey, view: &mut V) -> Result<()> {
+    pub async fn exec(&self, app: &mut App) -> Result<()> {
+        let fvk = app.fvk.clone();
         let hash = self
             .hash
             // We have to convert to uppercase because `tendermint::Hash` only accepts uppercase :(
             .to_uppercase()
             .parse()
             .context("invalid transaction hash")?;
-        // Retrieve Transaction
-        let tx = view.transaction_by_hash(hash).await?.ok_or_else(|| {
-            anyhow::anyhow!("transaction {} not found in view service", self.hash,)
-        })?;
+
+        // Retrieve Transaction from the view service first, or else the fullnode
+        let tx = if let Some(tx) = app.view().transaction_by_hash(hash).await? {
+            tx
+        } else {
+            println!("Transaction not found in view service, fetching from fullnode...");
+            // Fall back to fetching from fullnode
+            let mut client = app.tendermint_proxy_client().await?;
+            let rsp = client
+                .get_tx(GetTxRequest {
+                    hash: hex::decode(self.hash.clone())?,
+                    prove: false,
+                })
+                .await?;
+
+            let rsp = rsp.into_inner();
+            let tx = Transaction::decode(rsp.tx.as_slice())?;
+            tx
+        };
```

Comment:
Question: should this code be here, or should it be part of the view server implementation?

i.e., should we make the view server's transaction lookup query a fullnode if the transaction isn't found locally?

I think that that might be a bit cleaner than having to have this fallback code in (every) user of the view service.

Reply:
I think I agree with this. Another option is making a flag to enable/disable this behavior?

Reply:
Would that flag be in the request, or in the `pclientd` config? It seems better to have in in each individual request, like adding a
```
boolean fetch_if_missing = FIELDNUM;
```
to the request object